projects/plugins/jetpack/_inc/lib/class.jetpack-keyring-service-helper.php

```diff
@@ -180,29 +222,46 @@ static function admin_page_load() {
 					$verification = ( new Secrets() )->generate( 'publicize' );
 					if ( ! $verification ) {
 						$url = Jetpack::admin_url( 'jetpack#/settings' );
-						wp_die( sprintf( __( "Jetpack is not connected. Please connect Jetpack by visiting <a href='%s'>Settings</a>.", 'jetpack' ), $url ) );
+						wp_die(
+							sprintf(
+								wp_kses(
+									/* Translators: placeholder is a URL to a Settings page. */
+									__( "Jetpack is not connected. Please connect Jetpack by visiting <a href='%s'>Settings</a>.", 'jetpack' ),
+									array(
+										'a' => array(
+											'href' => array(),
+										),
+									)
+								),
+								esc_url( $url )
+							)
+						);
 
 					}
 					$stats_options = get_option( 'stats_options' );
 					$wpcom_blog_id = Jetpack_Options::get_option( 'id' );
 					$wpcom_blog_id = ! empty( $wpcom_blog_id ) ? $wpcom_blog_id : $stats_options['blog_id'];
 
 					$user     = wp_get_current_user();
-					$redirect = Jetpack_Keyring_Service_Helper::api_url( $service_name, urlencode_deep( array(
-						'action'       => 'request',
-						'redirect_uri' => add_query_arg( array( 'action' => 'done' ), menu_page_url( 'sharing', false ) ),
-						'for'          => 'publicize',
-						// required flag that says this connection is intended for publicize
-						'siteurl'      => site_url(),
-						'state'        => $user->ID,
-						'blog_id'      => $wpcom_blog_id,
-						'secret_1'     => $verification['secret_1'],
-						'secret_2'     => $verification['secret_2'],
-						'eol'          => $verification['exp'],
-					) ) );
-					wp_redirect( $redirect );
+					$redirect = self::api_url(
+						$service_name,
+						urlencode_deep(
+							array(
+								'action'       => 'request',
+								'redirect_uri' => add_query_arg( array( 'action' => 'done' ), menu_page_url( 'sharing', false ) ),
+								'for'          => 'publicize',
+								// required flag that says this connection is intended for publicize.
+								'siteurl'      => site_url(),
+								'state'        => $user->ID,
+								'blog_id'      => $wpcom_blog_id,
+								'secret_1'     => $verification['secret_1'],
+								'secret_2'     => $verification['secret_2'],
+								'eol'          => $verification['exp'],
+							)
+						)
+					);
+					wp_redirect( $redirect ); // phpcs:ignore WordPress.Security.SafeRedirect.wp_redirect_wp_redirect -- The API URL is an external URL and is filterable.
 					exit;
-					break;
 
 				case 'completed':
```

Comment:
There's no nonce check in this case. Should there be one? It is updating an option after all. Or is it ok because the option is just a cache the user has no control over (is that true?)?